eng/pipelines/templates/jobs/archetype-sdk-data.yml

```diff
@@ -0,0 +1,222 @@
+          JavaVersion: '1.11'
+
+    steps:
+      - script: |
+          echo "##vso[build.addbuildtag]Scheduled"
+        displayName: 'Tag scheduled builds'
+        condition: and(succeeded(), eq(variables['Build.SourceBranchName'],'master'),eq(variables['Build.Reason'],'Schedule'))
+
+      - task: UsePythonVersion@0
+        displayName: 'Use Python 3.6'
+        inputs:
+          versionSpec: '3.6'
+
+      - template: tools/daily-dev-build-variable/daily-dev-build-variable.yml@azure-sdk-tools
+
+      - ${{ each artifact in parameters.Artifacts }}:
+        - script: |
+            python3 --version
+            python3 eng/versioning/set_versions.py --build-type ${{parameters.SDKType}} --build-qualifier dev.$(Build.BuildNumber) --artifact-id ${{artifact.name}} --group-id ${{artifact.groupId}}
+          condition: and(succeeded(), eq(variables['SetDevVersion'],'true'))
+          displayName: Append dev package version suffix for ${{artifact.name}}
+
+      - script: |
+          python3 --version
+          python3 eng/versioning/update_versions.py --update-type library --build-type ${{parameters.SDKType}}
+        condition: and(succeeded(), eq(variables['SetDevVersion'],'true'))
+        displayName: Apply version settings to repository
+
+      - script: |
+          pip install markdown2 BeautifulSoup4
+        displayName: 'pip install markdown2 and BeautifulSoup4'
+
+      - template: ../steps/cache-maven-repository.yml
+
+      - template: eng/pipelines/templates/scripts/replace-relative-links.yml@azure-sdk-tools
+        parameters:
+          TargetFolder: $(Build.SourcesDirectory)
+          RootFolder: $(Build.SourcesDirectory)
+          BuildSHA: $(Build.SourceVersion)
+          RepoId: 'Azure/azure-sdk-for-java'
+
+      - task: Maven@3
+        displayName: 'Build and Package'
+        inputs:
+          mavenPomFile: sdk/${{parameters.ServiceDirectory}}/pom.service.xml
+          goals: 'package'
+          options: '$(DefaultOptions) "-DpackageOutputDirectory=$(Build.ArtifactStagingDirectory)" -DskipTests -Dgenerate-overview' # We include template-module so we ensure it always builds in CI
+          mavenOptions: '$(MemoryOptions) $(LoggingOptions)'
+          javaHomeOption: 'JDKVersion'
+          jdkVersionOption: $(JavaVersion)
+          jdkArchitectureOption: 'x64'
+          publishJUnitResults: false
+
+      - publish: $(Build.ArtifactStagingDirectory)
+        displayName: 'Publish Artifacts $(ArtifactName)'
+        artifact: '$(ArtifactName)'
+
+      - publish: $(System.DefaultWorkingDirectory)
+        condition: and(eq(variables['CaptureRepositoryOnFailure'], 'true'), failed())
+        displayName: 'Capture repo state for analysis'
+        artifact: repository
+
+  - job: 'Analyze'
+    condition: and(succeeded(), ne(variables['Skip.Analyze'], 'true'))
+
+    variables:
+      - template: ../variables/globals.yml
+
+    pool:
+      vmImage: 'ubuntu-16.04'
+
+    steps:
+      - pwsh: |
```

Comment:
I didn't look super close but it still feels like there might be a fair amount of duplication between the client and data templates. We should consider creating more templates to share as much as possible were it makes sense.